diesel/src/types/ops.rs

```diff
@@ -107,22 +107,30 @@ macro_rules! numeric_type {
 
 numeric_type!(SmallInt, Integer, BigInt, Float, Double, Numeric);
 
-impl Add for super::Timestamp {
-    type Rhs = super::Interval;
-    type Output = super::Timestamp;
-}
+macro_rules! interval_type {
+    ($($tpe: ident),*) => {
+        $(
+        impl Add for super::$tpe {
+            type Rhs = super::Interval;
+            type Output = super::$tpe;
+        }
 
-impl Sub for super::Timestamp {
-    type Rhs = super::Interval;
-    type Output = super::Timestamp;
-}
+        impl Add for super::Nullable<super::$tpe> {
+            type Rhs = super::Nullable<super::Interval>;
+            type Output = super::Nullable<super::$tpe>;
+        }
 
-impl Add for super::Date {
-    type Rhs = super::Interval;
-    type Output = super::Timestamp;
-}
+        impl Sub for super::$tpe {
+            type Rhs = super::Interval;
+            type Output = super::$tpe;
+        }
 
-impl Sub for super::Date {
-    type Rhs = super::Interval;
-    type Output = super::Timestamp;
+        impl Sub for super::Nullable<super::$tpe> {
+            type Rhs = super::Nullable<super::Interval>;
+            type Output = super::Nullable<super::$tpe>;
```

Comment:
While writing a test for that, I noticed the output types are always Timestamp here: https://www.postgresql.org/docs/9.6/static/functions-datetime.html